pkg/list/list_components.go

```diff
@@ -25,17 +30,80 @@ func getStackComponents(stackData any) ([]string, error) {
 		return nil, fmt.Errorf("could not parse Terraform components")
 	}
 
-	return lo.Keys(terraformComponents), nil
+	uniqueKeys := lo.Keys(terraformComponents)
+	result := make([]string, 0)
+
+	for _, dataKey := range uniqueKeys {
+		data := terraformComponents[dataKey]
+		rowData := make([]string, 0)
+		for _, key := range listFields {
+			value, found := resolveKey(data.(map[string]any), key)
+			if !found {
+				value = "-"
+			}
+			rowData = append(rowData, fmt.Sprintf("%s", value))
+		}
+		result = append(result, strings.Join(rowData, "\t\t"))
+	}
+	return result, nil
+}
+
+// resolveKey resolves a key from a map, supporting nested keys with dot notation
+func resolveKey(data map[string]any, key string) (any, bool) {
+	// Remove leading dot from the key (e.g., `.vars.tenant` -> `vars.tenant`)
+	key = strings.TrimPrefix(key, ".")
+
+	// Split key on `.`
+	parts := strings.Split(key, ".")
+	current := data
+
+	// Traverse the map for each part
+	for i, part := range parts {
+		if i == len(parts)-1 {
+			// Return the value for the last part
+			if value, exists := current[part]; exists {
+				return value, true
+			}
+			return nil, false
+		}
+
+		// Traverse deeper
+		if nestedMap, ok := current[part].(map[string]any); ok {
+			current = nestedMap
+		} else {
+			return nil, false
+		}
+	}
+
+	return nil, false
 }
 
 // FilterAndListComponents filters and lists components based on the given stack
-func FilterAndListComponents(stackFlag string, stacksMap map[string]any) (string, error) {
+func FilterAndListComponents(stackFlag string, stacksMap map[string]any, listConfig schema.ListConfig) (string, error) {
 	components := []string{}
 
+	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', tabwriter.AlignRight)
+	header := make([]string, 0)
+	listFields := make([]string, 0)
+
+	re := regexp.MustCompile(`\{\{\s*(.*?)\s*\}\}`)
+
+	for _, v := range listConfig.Columns {
+		header = append(header, v.Name)
+		match := re.FindStringSubmatch(v.Value)
+
+		if len(match) > 1 {
+			listFields = append(listFields, match[1])
+		} else {
+			return "", fmt.Errorf("invalid value format for column name %s", v.Name)
+		}
+	}
+	fmt.Fprintln(writer, strings.Join(header, "\t\t"))
+
```

Comment:
Rewrite to use lipgloss

Reply:
yes, on it

Reply:
![image](https://github.com/user-attachments/assets/23f3a966-3bba-4f00-96ce-339ea0343b61)

using lipgloss. For folder value, we just need to pass the correct value in the yaml.